### DISCUSSION TITLE
Problem controlling button previous get weird reaction

### DISCUSSION THREAD

**Discussion by moddingmaster** [tomaskovacik/vwcdavr#20] [General]
Hi, in advance i have to say that is marvelous work, congrats.
However i have one hickup in one function, all works well, but one previous function makes some weird reaction when is past 10 seconds on radio timer, if button press is beyond 10s, the controller just presses prev and next making music restart and then goes to the next one.
I noticed that, on the sketch, it refers to previous button timers (line 464 of cdc_arduino_ported_k9spud_vwcdpic_headphone_support.ino), being double, maybe it is the problem, but i cant find the function where reacts to that doubling instead of single timers like next or play buttons.

Thank you, and for your time.

**Comment by tomaskovacik** [comment]
which sketch are you using? post link here, please

**Reply by moddingmaster** [reply]
Is this one:
https://github.com/tomaskovacik/vwcdavr/blob/master/CDC_emulator/cdc_arduino_ported_k9spud_vwcdpic_headphone_support/cdc_arduino_ported_k9spud_vwcdpic_headphone_support.ino

**Comment by tomaskovacik** [comment]
strange 
 //this is what is fired for previous button,
`  if (prev_count > 0) {//it is fired each 50ms until prev_count is = 0
    Serial.println("prev"); //debug, you can remove it
    Serial.println(play_count); //debug, you can remove it
    Serial.println(next_count); //debug, you can remove it
    Serial.println(prev_count); //debug, you can remove it
    prev_count--;//decrement variable prev_count

#if defined(BUTTONS_ACTIVE_HIGH)  //button push pull pin up or down based on DEFINE
    ANDROID_PREV_PORT |= _BV(ANDROID_PREV); //high //PORTD |= B10000000; //digitalWrite(7, HIGH);
#elif defined(BUTTONS_ACTIVE_LOW)
    ANDROID_PREV_PORT &= ~_BV(ANDROID_PREV); //low //PORTD &= B01111111; //digitalWrite(7, LOW);
#endif
  }  else { // if button == 0 we flip pin state 
#if defined(BUTTONS_ACTIVE_HIGH)
    ANDROID_PREV_PORT &= ~_BV(ANDROID_PREV); //low //PORTD &= B01111111; //digitalWrite(7, LOW);
#elif defined(BUTTONS_ACTIVE_LOW)
    ANDROID_PREV_PORT |= _BV(ANDROID_PREV); //high //PORTD |= B10000000; //digitalWrite(7, HIGH);
#endif

    //uncomment in case smartphone/MP3 player requires double press to go to previous song
    /*
      if (prev_count_delay > 0 ) { //we ended first push and do not finished second push
      prev_count_delay--;
      if (prev_count_delay == 0) //we are at last run of delay loop
        prev_count = ANDROID_PUSH_COUNT;
      }
    */
  }`

so I do not really see wtf :)) 

only chance is debug it via serial port unfortunately :(

**Comment by moddingmaster** [comment]
i have debugged the data from my RNS-E, as in fact, isn't the code with problem, it is the radio that sends prev and then next. Strange behaviour, as it is different from the chorus II and concert. 
Maybe i will add a 1s inhibitor after prev data and send that to you if anyone else has that same problem.
Sorry and thank you for the time spent in my problem that isn't really a problem.
